src/components/v2/V2Project/ProjectActivity/eventElems/DistributeReservedTokensElem.tsx

```diff
@@ -0,0 +1,149 @@
+import { Trans } from '@lingui/macro'
+import { smallHeaderStyle } from 'components/shared/activityEventElems/styles'
+import EtherscanLink from 'components/shared/EtherscanLink'
+import FormattedAddress from 'components/shared/FormattedAddress'
+import { ThemeContext } from 'contexts/themeContext'
+import { V1ProjectContext } from 'contexts/v1/projectContext'
+import useSubgraphQuery from 'hooks/SubgraphQuery'
+import { DistributeReservedTokensEvent } from 'models/subgraph-entities/v2/distribute-reserved-tokens-event'
+import { useContext } from 'react'
+import { formatHistoricalDate } from 'utils/formatDate'
+import { formatWad } from 'utils/formatNumber'
+import { tokenSymbolText } from 'utils/tokenSymbolText'
+
+export default function DistributeReservedTokensEventElem({
+  event,
+}: {
+  event:
+    | Pick<
+        DistributeReservedTokensEvent,
+        | 'id'
+        | 'timestamp'
+        | 'txHash'
+        | 'caller'
+        | 'beneficiary'
+        | 'beneficiaryTokenCount'
+        | 'tokenCount'
+      >
+    | undefined
+}) {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+  const { tokenSymbol } = useContext(V1ProjectContext)
+
+  const { data: distributeEvents } = useSubgraphQuery(
+    {
+      entity: 'distributeToReservedTokenSplitEvent',
+      keys: [
+        'id',
+        'timestamp',
+        'txHash',
+        'beneficiary',
+        'tokenCount',
+        'projectId',
+      ],
+      orderDirection: 'desc',
+      orderBy: 'tokenCount',
+      where: event?.id
+        ? {
+            key: 'distributeReservedTokensEvent',
+            value: event.id,
+          }
+        : undefined,
+    },
+    {},
+  )
+
+  if (!event) return null
+
+  return (
+    <div>
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-between',
+        }}
+      >
+        <div style={smallHeaderStyle(colors)}>
+          <Trans>Distributed reserved</Trans>{' '}
+          {tokenSymbolText({
+            tokenSymbol: tokenSymbol,
+            capitalize: false,
+            plural: true,
+          })}
+        </div>
+
+        <div style={{ textAlign: 'right' }}>
+          <div style={smallHeaderStyle(colors)}>
+            {event.timestamp && (
+              <span>{formatHistoricalDate(event.timestamp * 1000)}</span>
+            )}{' '}
+            <EtherscanLink value={event.txHash} type="tx" />
+          </div>
+          <div style={smallHeaderStyle(colors)}>
+            <Trans>called by</Trans> <FormattedAddress address={event.caller} />
```

Comment:
same here.

**suggestion (non-blocking)** maybe we can consider a shared component for these to avoid duplication.